### PR TITLE
Feature/label

### PR DESCRIPTION
**Pull request by JadhavJeet**

### CODE REVIEW

**Review by Ibabalola — APPROVED** [review]

File: stories/Label/Documentation.mdx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+<Canvas of={LabelStories.Unstyled} />
+
+A more complex example with all the possible properties is:
+
+```js
+<Label
+  className="dcx-label additional classes"
+  value="text"
+  id="user-defined-id"
```
Comment (daniele-zurico) [inline stories/Label/Documentation.mdx:20]: I will add as well props... for example
```html
<Label
  className="dcx-label additional classes"
  value="text"
  id="user-defined-id"
  props={{}}
/>
```

File: stories/liveEdit/LabelLive.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live';
+import { Label } from '../../src/label/Label';
+
+const LabelDemo = `
+function LabelDemo() {
+
+  return (
+    <Label
+      className="dcx-label additional classes" value="text" id="user-defined-id"
```
Comment (daniele-zurico) [inline stories/liveEdit/LabelLive.tsx:10]: add props={{}} remove dcx-label because is not user defined

File: src/label/Label.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  className?: string;
+  /**
+   * Allows us to use the Value in label
+   */
+  value?: string;
+  /**
+   * it will pass an id to the label element
+   */
+  id?: string;
+};
```
Comment (daniele-zurico) [inline src/label/Label.tsx:17]:
```suggestion
     /**
     * Additional props/attributes
     */
    props?: React.HTMLAttributes<HTMLHeadingElement>;
};
```

File: src/label/Label.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+   */
+  className?: string;
+  /**
+   * Allows us to use the Value in label
+   */
+  value?: string;
+  /**
+   * it will pass an id to the label element
+   */
+  id?: string;
```
Comment (daniele-zurico) [inline src/label/Label.tsx:16]:
```suggestion
     /**
     * Additional props/attributes
     */
    props?: React.HTMLAttributes<HTMLHeadingElement>;
```

File: src/label/Label.tsx (hunk 3 above)
Comment (daniele-zurico) [inline src/label/Label.tsx:17]:
```ts 
     /**
     * Additional props/attributes
     */
    props?: React.HTMLAttributes<HTMLHeadingElement>;
}
```

File: src/label/Label.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+export const Label = ({ className, value, id, ...props }: labelProps) => (
```
Comment (daniele-zurico) [inline src/label/Label.tsx:19]:
```suggestion
export const Label = ({ className, value, id, props }: labelProps) => (
```

**Review by daniele-zurico — COMMENTED** [review]
if you rebase your branch with `release/0.8` you'll see the first tipography component created... 
you need to update the file `.storybook/preview.js`:
```
          'Typography', [
            'Heading',['Documentation', 'Live', 'Without style', 'Class based']
          ],
```
adding your Label.... something like:
```
 'Heading',['Documentation', 'Live', 'Without style', 'Class based'],
 'Label',['Documentation', 'Live', 'Without style', 'Class based']
```

File: src/label/Label.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+type labelProps = {
+  /**
+   * A CSS class for styling label
+   */
+  className?: string;
+  /**
+   * Allows us to use the Value in label
+   */
+  value?: string;
```
Comment (daniele-zurico) [inline src/label/Label.tsx:12]: I suppose value is mandatory for a label

File: stories/Label/ClassBased.stories.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    },
+  },
+  tags: ['autodocs'],
+};
+
+export const Basic = {
+  name: 'Basic',
+  args: {
+    value: 'text',
+    className: 'govuk-button',
```
Comment (daniele-zurico) [inline stories/Label/ClassBased.stories.js:22]: use `govuk-label` that will make more sense.

File: stories/liveEdit/LabelLive.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live';
+import { Label } from '../../src/label/Label';
+
+const LabelDemo = `
+function LabelDemo() {
+
+  return (
+    <Label
+      className="additional classes" value="text" id="user-defined-id" props={{}}
```
Comment (daniele-zurico) [inline stories/liveEdit/LabelLive.tsx:10]: remove `classes"` please

File: stories/liveEdit/LabelLive.tsx (hunk 2 above)
Comment (daniele-zurico) [reply stories/liveEdit/LabelLive.tsx]: Despite it shows as outdated I still can see that `dcx-label` is there and there's not props={{}}

File: stories/Label/ClassBased.stories.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import { Label } from '../../src/label/Label';
+
+/**
+ * In this section we're using the Label component providing the **GovUk style** passing the relative className.
+ * Feel free to use your own css to style the formInput as you prefer.
+ */
+export default {
+  title: 'DCXLibrary/Form/Label/Class based',
```
Comment (daniele-zurico) [inline stories/Label/ClassBased.stories.js:8]:
```suggestion
  title: 'DCXLibrary/Typography/Label/Class based',
```

File: stories/Label/Documentation.mdx [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import { Meta, Canvas, ArgTypes } from '@storybook/blocks';
+import * as LabelStories from './UnStyled.stories';
+
+<Meta title="DCXLibrary/Form/Label/Documentation" />
```
Comment (daniele-zurico) [inline stories/Label/Documentation.mdx:4]:
```suggestion
<Meta title="DCXLibrary/Typography/Label/Documentation" />
```

File: stories/Label/Live.stories.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import { Label } from '../../src/label/Label';
+import LabelLive from '../liveEdit/LabelLive';
+
+export default {
+  title: 'DCXLibrary/Form/Label/Live',
```
Comment (daniele-zurico) [inline stories/Label/Live.stories.js:5]:
```suggestion
  title: 'DCXLibrary/Typography/Label/Live',
```

File: stories/Label/UnStyled.stories.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import { Label } from '../../src/label/Label';
+
+export default {
+  title: 'DCXLibrary/Form/Label/Without style',
```
Comment (daniele-zurico) [inline stories/Label/UnStyled.stories.js:4]:
```suggestion
  title: 'DCXLibrary/Typography/Label/Without style',
```

File: src/label/__test__/Label.test.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    expect(container.querySelector('.dcx-label')).toBeInTheDocument();
+    expect(container.querySelector('.styleOne')).toBeInTheDocument();
+  });
+
+  it('should be able to pass an id', () => {
+    const { container } = render(<Label value="text" id="my_id" />);
+    expect(container.firstChild).toHaveAttribute('id', 'my_id');
+  });
+
+  it('should be able to pass some extra properties', async () => {
```
Comment (daniele-zurico) [inline src/label/__test__/Label.test.tsx:31]:
```suggestion
  it('should be able to pass some extra properties', () => {
```

**Review by daniele-zurico — APPROVED** [review]

**Pull request merged**